### ISSUE TITLE
library "libobjc.so.4.6" not found

### ISSUE THREAD

**Issue by kloned** [crystax/android-samples-opengl#3]
Now, I have trouble running the resulting apk.
I've used https://dl.crystax.net/builds/809/darwin/ (the latest build available for OSX)

Application crashes on device with following log:

```
("/data/app/net.crystax.sample.opengl-1/lib/arm/libCoreServices.so", RTLD_LAZY) failed: dlopen failed: library "libobjc.so.4.6" not found
E/AndroidRuntime( 9256): FATAL EXCEPTION: main
E/AndroidRuntime( 9256): Process: net.crystax.sample.opengl, PID: 9256
E/AndroidRuntime( 9256): java.lang.UnsatisfiedLinkError: dlopen failed: library "libobjc.so.4.6" not found
E/AndroidRuntime( 9256):    at java.lang.Runtime.loadLibrary(Runtime.java:372)
E/AndroidRuntime( 9256):    at java.lang.System.loadLibrary(System.java:988)
E/AndroidRuntime( 9256):    at net.crystax.sample.opengl.OpenGLApplication.<clinit>(OpenGLApplication.java:7)
E/AndroidRuntime( 9256):    at java.lang.reflect.Constructor.newInstance(Native Method)
E/AndroidRuntime( 9256):    at java.lang.Class.newInstance(Class.java:1690)
E/AndroidRuntime( 9256):    at android.app.Instrumentation.newApplication(Instrumentation.java:1004)
E/AndroidRuntime( 9256):    at android.app.Instrumentation.newApplication(Instrumentation.java:989)
E/AndroidRuntime( 9256):    at android.app.LoadedApk.makeApplication(LoadedApk.java:659)
E/AndroidRuntime( 9256):    at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5965)
E/AndroidRuntime( 9256):    at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method)
E/AndroidRuntime( 9256):    at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:693)
E/AndroidRuntime( 9256):    at android.app.ActivityThread.handleBindApplication(<Xposed>)
E/AndroidRuntime( 9256):    at android.app.ActivityThread.access$1700(ActivityThread.java:197)
E/AndroidRuntime( 9256):    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1748)
E/AndroidRuntime( 9256):    at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 9256):    at android.os.Looper.loop(Looper.java:145)
E/AndroidRuntime( 9256):    at android.app.ActivityThread.main(ActivityThread.java:6897)
E/AndroidRuntime( 9256):    at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 9256):    at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime( 9256):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
E/AndroidRuntime( 9256):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
E/AndroidRuntime( 9256):    at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:117)
```

**Comment by kloned** [comment]
The same thing with build 799 mentioned in readme

**Comment by dmsck** [comment]
Yes, for some reason libobjc.so included to darwin NDK build was built with SONAME set to "libobjc.so.4.6", while it should be "libobjc.so". This is reason why it can't be loaded on device. This is somehow related to the recently introduced CMake build... Thank you for reporting, I'm going to fix it.

**Comment by kloned** [comment]
Hi, thank you for all your hard work!
Just a question, somehow referring to actual project. 
Is it planned crystax/objective-c integration with https://github.com/godotengine/godot ?
(It's under MIT license, multiplatform 2D/3D lightweight game engine, with its own editor, including GUI support.)
Thanks

**Comment by dmsck** [comment]
@kloned No, it's not planned, but contributions are welcome! However, this question is off-topic here, so if you want to continue conversation, please use our mailing group (https://groups.google.com/forum/#!forum/crystax-ndk).

**Comment by kloned** [comment]
I apologize for off-topic and thanks for link to general discussion forum.
The problem of this thread is still remain, though :)

**Comment by dmsck** [comment]
This bug is fixed in build [#845](https://dl.crystax.net/builds/845/)
